x/auth/ante/sigverify.go

```diff
@@ -447,3 +472,40 @@ func CountSubKeys(pub cryptotypes.PubKey) int {
 
 	return numKeys
 }
+
+// signatureDataToBz converts a SignatureData into raw bytes signature. It is
+// the same function as in auth/tx/sigs.go, but copied here because of import
+// cycles.
+func signatureDataToBz(data signing.SignatureData) ([]byte, error) {
```

Comment:
We have `x/auth/tx` imports `x/auth/ante.HasExtensionOptionsTx`. Which means I cannot use `x/auth/tx.SignatureDataToModeInfoAndSig` here, and need to duplicate code...

IMO, the cleanest would be to move `HasExtensionOptionsTx` from `ante` to `tx`, so that we have `ante` dependending on `tx` and not the other way round. But that's an API-breaking change.

any other ideas?

Reply:
Ahh yes. Are we trying to backport this? We can keep this as is and once backported, we can make a subsequent PR that does your more ideal change to master. WDYT?

Reply:
Yeah, we're backporting to both 0.42 and 0.43. The follow-up PR sgtm: https://github.com/cosmos/cosmos-sdk/issues/9753.